packages/util/index.d.ts

```diff
@@ -15,3 +16,43 @@ export class EnsureSafeComponentHelper<
   };
   Return: C extends string ? ComponentLike<unknown> : C;
 }> {}
+
+export class macroGetConfig extends Helper<{
+  Args: { Positional: Parameters<typeof getConfig>; };
+  Return: ReturnType<typeof getConfig>;
+}> {}
+
+export class macroGetOwnConfig extends Helper<{
+  Args: { Positional: Parameters<typeof getOwnConfig>; };
+  Return: ReturnType<typeof getOwnConfig>;
+}> {}
+
+export class macroCondition extends Helper<{
+  Args: { Positional: [predicate: unknown] };
+  Return: boolean;
+}> {}
+
+export class macroDependencySatisfies extends Helper<{
+  Args: { Positional: Parameters<typeof dependencySatisfies>; };
+  Return: ReturnType<typeof dependencySatisfies>;
+}> {}
+
+export class macroMaybeAttrs extends Helper<{
+  Args: { Positional: Parameters<typeof maybeAttrs>; };
+  Return: ReturnType<typeof maybeAttrs>;
+}> {}
+
+export class macroFailBuild extends Helper<{
+  Args: { Positional: Parameters<typeof failBuild>; };
+  Return: ReturnType<typeof failBuild>;
+}> {}
+
+export interface EmbroiderMacrosRegistry {
+  'ensure-safe-component': typeof EnsureSafeComponentHelper;
+  macroGetOwnConfig: typeof macroGetOwnConfig;
+  macroGetConfig: typeof macroGetConfig;
+  macroCondition: typeof macroCondition;
+  macroDependencySatisfies: typeof macroDependencySatisfies;
+  macroMaybeAttrs: typeof macroMaybeAttrs;
+  macroFailBuild: typeof macroFailBuild;
+}
```

Comment:
These types look good for the template macros but they should go in `@embroider/macros` not `@embroider/util`.

Reply:
ok, moved these to packages/macros/src/template-registry.ts, not sure if that's the right spot, but it didn't seem like index.ts would be a proper fit.

Also had to rename all helpers definitions to PascalCase because of an ESLint rule, don't think it matters, registry names them in camelCase as they appear in templates.